Objects/typeobject.c

```diff
@@ -3875,7 +3875,7 @@ static int
 type_setattro(PyTypeObject *type, PyObject *name, PyObject *value)
 {
     int res;
-    if (!(type->tp_flags & Py_TPFLAGS_HEAPTYPE)) {
```

Comment:
If I am not missing anything,t his change will alter external code that has `Py_TPFLAGS_HEAPTYPE` set

Reply:
Hm, yes it breaks ABI compatibility. This should be sufficient:

```suggestion
    if (!(type->tp_flags & Py_TPFLAGS_HEAPTYPE) ||
        type->tp_flags & Py_TPFLAGS_IMMUTABLETYPE)
    {
```

Reply:
...or does it? `PyType_Ready` sets `Py_TPFLAGS_IMMUTABLETYPE` for all static types.

Reply:
Is not about ABI compatibility, as about changing the behaviour of classes that have `Py_TPFLAGS_HEAPTYPE`